src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadonlyStructTests.cs

```diff
@@ -1308,6 +1308,211 @@ public static void Main()
+
+                Assert.False(peModule.Module.HasIsReadOnlyAttribute(((PEPropertySymbol)p1).Handle));
+                Assert.True(peModule.Module.HasIsReadOnlyAttribute(((PEMethodSymbol)p1.GetMethod).Handle));
+                Assert.False(peModule.Module.HasIsReadOnlyAttribute(((PEMethodSymbol)p1.SetMethod).Handle));
+
+                Assert.True(peModule.Module.HasIsReadOnlyAttribute(((PEPropertySymbol)p2).Handle));
+                Assert.True(peModule.Module.HasIsReadOnlyAttribute(((PEMethodSymbol)p2.GetMethod).Handle));
+
+                Assert.False(peModule.Module.HasIsReadOnlyAttribute(((PEPropertySymbol)p3).Handle));
+                Assert.True(peModule.Module.HasIsReadOnlyAttribute(((PEMethodSymbol)p3.GetMethod).Handle));
+
+                Assert.False(peModule.Module.HasIsReadOnlyAttribute(((PEPropertySymbol)p4).Handle));
+                Assert.True(peModule.Module.HasIsReadOnlyAttribute(((PEMethodSymbol)p4.SetMethod).Handle));
+
+                Assert.False(peModule.Module.HasIsReadOnlyAttribute(((PEPropertySymbol)p5).Handle));
+                Assert.False(peModule.Module.HasIsReadOnlyAttribute(((PEMethodSymbol)p5.GetMethod).Handle));
+                Assert.False(peModule.Module.HasIsReadOnlyAttribute(((PEMethodSymbol)p5.SetMethod).Handle));
+
+                AssertDeclaresType(peModule, WellKnownType.System_Runtime_CompilerServices_IsReadOnlyAttribute, Accessibility.Internal);
+            }
+        }
+
+        [Fact]
+        public void ReadOnlyStruct_ReadOnlyMembers_Metadata()
+        {
+            var csharp = @"
+public readonly struct S
+{
+    public void M1() {}
+    public readonly void M2() {}
+
+    public int P1 { get; }
+    public readonly int P2 => 42;
+    public int P3 { readonly get => 123; }
+    public int P4 { readonly set {} }
+    public static int P5 { get; set; }
+}
+";
+            CompileAndVerify(csharp, symbolValidator: validate);
+
+            void validate(ModuleSymbol module)
+            {
+                var type = module.ContainingAssembly.GetTypeByMetadataName("S");
+                var m1 = type.GetMethod("M1");
+                var m2 = type.GetMethod("M2");
+
+                var p1 = type.GetProperty("P1");
+                var p2 = type.GetProperty("P2");
+                var p3 = type.GetProperty("P3");
+                var p4 = type.GetProperty("P4");
+                var p5 = type.GetProperty("P5");
+
+                var peModule = (PEModuleSymbol)module;
+
+                Assert.False(peModule.Module.HasIsReadOnlyAttribute(((PEMethodSymbol)m1).Handle));
+                Assert.True(peModule.Module.HasIsReadOnlyAttribute(((PEMethodSymbol)m2).Handle));
+
+                Assert.False(peModule.Module.HasIsReadOnlyAttribute(((PEPropertySymbol)p1).Handle));
+                Assert.True(peModule.Module.HasIsReadOnlyAttribute(((PEMethodSymbol)p1.GetMethod).Handle));
+
+                Assert.True(peModule.Module.HasIsReadOnlyAttribute(((PEPropertySymbol)p2).Handle));
```

Comment:
>Assert.True(peModule.Module.HasIsReadOnlyAttribute(((PEPropertySymbol)p2).Handle)); [](start = 16, length = 83)

Is there value in this scenario? I wonder if we should disallow readonly member in readonly struct (since redundant). 
Same thing for a readonly getter (or setter) in a readonly property.

Reply:
After some offline discussion we concluded that properties in metadata should never use `IsReadOnlyAttribute` to indicate that the accessors are `readonly`.

Reply:
Is it necessary to disallow `readonly` for members of a `readonly struct`? `readonly` is already allowed (and required) for fields in a `readonly struct`.

---
In reply to: [267476750](https://github.com/dotnet/roslyn/pull/34260#discussion_r267476750) [](ancestors = 267476750)

Reply:
`readonly` field in a `readonly struct` may make sense if the field is public. Illustrative scenario below.
On the other hand, `readonly` methods in a `readonly struct` seem purely redundant. 
We could also leave that to the IDE to offer a fix (remove redundant readonly)....

```C#
public readonly struct S 
{
    public readonly int i;
}
public class D
{
    void M(S s)
    {
        s.i = 10;  // error CS0191: A readonly field cannot be assigned to (except in a constructor or a variable initializer)
    }
}
```

---
In reply to: [267820508](https://github.com/dotnet/roslyn/pull/34260#discussion_r267820508) [](ancestors = 267820508,267476750)